### PR TITLE
 Add benchmark for indexing with .loc for sorted/unsorted DatetimeIndex

### LINKED ISSUE #38690: ASV: Add benchmark for indexing with .loc for sorted/unsorted DatetimeIndex
xref #17754, a benchmark should be added for indexing with .loc for sorted/unsorted `DatetimeIndex`. With a constructed `DatetimeIndex`, for example

```python
dates = pd.date_range('2011-1-1', periods=500000, freq='min')
index = np.random.choice(dates, 500000, replace=True)
df = pd.DataFrame(index=index, data={'a': 1})
df_sorted = df.sort_index()
```

the behavior of interest to benchmark is
```python
df.loc['2011-6-11']
```
and 
```python
df_sorted.loc['2011-6-11']
```

### PR DESCRIPTION
**Pull request by weikhor**
- [x] closes #38690 
- [x] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [x] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [ ] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.

### CODE REVIEW

**Comment by weikhor** [comment]
@mroeschke Hi The testing pipeline in green. Thank for review.

File: asv_bench/benchmarks/indexing.py [hunk 1]
```diff
@@ -290,20 +290,25 @@ def setup(self):
...
         #  ser = Series(range(len(dti)), index=dti)
         #  ser[dti2]
         self.dti.get_indexer(self.dti2)
 
+
+class SortedAndUnsortedDatetimeIndexLoc:
+    def setup(self):
+        dti = date_range("2016-01-01", periods=10000, tz="US/Pacific")
+        index = np.array(dti)
+        df_unsorted = DataFrame(index=index, data={"a": 1})
```
[the remark this answers is not in the document]
Comment (weikhor) [reply asv_bench/benchmarks/indexing.py:304]: Have do sort testing
```
self.df_unsorted = df_unsorted
print(self.df_unsorted.index.is_monotonic_increasing)
```
```
False
```
```
self.df_sort = df_sort
print(self.df_sort.index.is_monotonic_increasing)
```
```
True
```

**Comment by weikhor** [comment]
@mroeschke Pipeline check in green

**Review by mroeschke — APPROVED** [review]

**Comment by mroeschke** [comment]
Thanks for following up @weikhor

**Pull request merged**